sqlglot/executor/python.py

```diff
@@ -142,15 +142,16 @@ def scan_csv(self, step):
             context = self.context({alias: table})
             yield context
             types = []
-
             for row in reader:
                 if not types:
                     for v in row:
                         try:
                             types.append(type(ast.literal_eval(v)))
                         except (ValueError, SyntaxError):
                             types.append(str)
-                context.set_row(tuple(t(v) for t, v in zip(types, row)))
+
+                # for empty string, set default value instead of casting
+                context.set_row(tuple(t(v) if v else t() for t, v in zip(types, row)))
```

Comment:
I don't think this could work because there are some aggregate queries and they are invalid with `None`:
```
WITH customer_total_return
     AS (SELECT sr_customer_sk     AS ctr_customer_sk,
                sr_store_sk        AS ctr_store_sk,
                Sum(sr_return_amt) AS ctr_total_return
         FROM   store_returns,
                date_dim
         WHERE  sr_returned_date_sk = d_date_sk
                AND d_year = 2001
         GROUP  BY sr_customer_sk,
                   sr_store_sk)
SELECT c_customer_id
FROM   customer_total_return ctr1,
       store,
       customer
WHERE  ctr1.ctr_total_return > (SELECT Avg(ctr_total_return) * 1.2
                                FROM   customer_total_return ctr2
                                WHERE  ctr1.ctr_store_sk = ctr2.ctr_store_sk)
       AND s_store_sk = ctr1.ctr_store_sk
       AND s_state = 'TN'
       AND ctr1.ctr_customer_sk = c_customer_sk
ORDER  BY c_customer_id
LIMIT 100
``` 

Error: `Step 'Aggregate: customer_total_return (4936004752)' failed: '<' not supported between instances of 'NoneType' and 'int'`

Reply:
that means that the aggregation code is not correct. they should be able to support None

Reply:
Oh ok, let me investigate more